EpikCord/client/commands.py

```diff
@@ -49,17 +52,455 @@ def to_dict(self):
+        required: bool = False,
+    ):
+        super().__init__(
+            name,
+            description,
+            name_localizations=name_localizations,
+            description_localizations=description_localizations,
+            required=required,
+            type=ApplicationCommandOptionType.MENTIONABLE,
+        )
+
+class ChannelOption(BaseApplicationCommandOption):
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        *,
+        name_localizations: List[Localization] = [],
+        description_localizations: List[Localization] = [],
+        required: bool = False,
+        channel_types: Optional[List[ChannelType]]
+    ):
+        super().__init__(
+            name,
+            description,
+            name_localizations=name_localizations,
+            description_localizations=description_localizations,
+            required=required,
+            type=ApplicationCommandOptionType.CHANNEL,
+        )
+        self.channel_types = channel_types
+
+    def to_dict(self):
+        payload = super().to_dict()
+        if self.channel_types:
+            payload["channel_types"] = self.channel_types
+        return payload
+
+ApplicationCommandOption = Union[
+    IntegerOption,
+    NumberOption,
+    StringOption,
+    BooleanOption,
+    UserOption,
+    RoleOption,
+    AttachmentOption,
+    MentionableOption,
+    ChannelOption,
+]
+
+
+class ClientChatInputCommand(BaseClientCommand):
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        callback: AsyncFunction,
+        *,
+        guild_ids: List[int] = [],
+        name_localizations: List[Localization] = [],
+        description_localizations: List[Localization] = [],
+        type: ApplicationCommandType = ApplicationCommandType.CHAT_INPUT,
+        guild_only: Optional[bool] = None,
+        default_member_permissions: Optional[Permissions] = None,
+        nsfw: bool = False,
+        options: Optional[List[str]] = None,
+    ):
+        super().__init__(
+            name,
+            callback,
+            guild_ids=guild_ids,
+            name_localizations=name_localizations,
+            type=type,
+            guild_only=guild_only,
+            default_member_permissions=default_member_permissions,
+            nsfw=nsfw,
+        )
+        self.description = description
+        self.description_localizations = description_localizations
+        self.options = options
+
+    def to_dict(self):
+
+        payload = super().to_dict()
+
+        payload["description"] = self.description
+
+        if description_localizations := localization_list_to_dict(
+            self.description_localizations
+        ):
+            payload["description_localizations"] = description_localizations
+        if self.options:
+            payload["options"] = [option.to_dict() for option in self.options] # type: ignore # I still need to create the Option class
+
+        return payload
```

Comment:
I think using dataclasses would have made this endeavor twice shorter

Reply:
How?

Reply:
```py
>>> @dataclass
... class Foo:
...     a: int
... 
>>> 
>>> @dataclass
... class Bar(Foo):
...     b: str
... 
>>> 
>>> Bar(b='owo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Bar.__init__() missing 1 required positional argument: 'a'
```

Reply:
> So how would that make our current process faster?

less code